### ISSUE TITLE
VS Code with React/NextJS Project- Throws flase positive on Link component with a tag

### ISSUE THREAD

**Issue by jennifert** [dequelabs/axe-core#3070]
Product:  axe Linter (VS Code)

Expectation: Axe will check the LINK href in next (ignoring a missing href when you use an a tag inside link component to set class names). In NextJS this is the [Official documented](https://nextjs.org/docs/api-reference/next/link) way to add a class for pages within projects. 

Actual: Throws errors that anchor tag href is missing: 
> Axe Linter (anchor-is-valid): Only use a elements as links, including a functional href attribute.

![Screen Shot 2021-07-08 at 5 33 43 PM](https://user-images.githubusercontent.com/3802757/124993647-a8259300-e012-11eb-914d-8053c90c3b1f.png)
[Code in screenshot](https://github.com/jennifert/nextjs-site/blob/main/components/nav.js)

Motivation: will be able to ensure link tag is properly filled in, instead of a false positive for the class override or ignoring the rule in the project (as links externally will use the a tag as expected).

<pre><code>
extension version: 4.2.1
VS code: 
Version: 1.57.1
Commit: 507ce72a4466fbb27b715c3722558bb15afa9f48
Date: 2021-06-17T13:28:32.912Z (3 wks ago)
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 20.5.0

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 16.4.2 (from my terminal)
- Platform:  Mac
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. I'll be sure to pass it along to the appropriate people.

**Comment by stephenmathieson** [comment]
This is tricky, as `<a>Click me</a>` _should_ produce violations, however, in Next.js, you wrap anchors (or other elements) with `<Link>`.

To solve this properly, we'd want to:
- check if `Link` is imported from `next/link`
- ensure anchors without `href`s are wrapped by a `Link` which does have an `href`
- ensure `Link`s aren't wrapping non-anchors (unless they're marked up as anchors (eg with `tabIndex` and `role=link`))

**Comment by jennifert** [comment]
Thanks, sorry was not sure how to word what was happening properly to show how Next is vs regular scenarios. Otherwise the linter extension worked great in the project.

I see what you mean with how it can be tricky. I am also using @axe-core/cli in my package.json, so that would catch any Link anchors missed in the editor.

 Thank you all :)

**Comment by stephenmathieson** [comment]
For now, you might want to disable the rule for your project:

**axe-linter.yml**
```yaml
rules:
  anchor-is-valid: false
```

As I'm not sure when we'll get a patch in to fix this.

**Comment by jennifert** [comment]
Will do, thanks again! 👍

**Comment by WilcoFiers** [comment]
We currently have a PR open which is going to remove this rule. Unfortunately this one isn't quite working out. The next release of axe Linter will no longer have this problem.
